### ISSUE TITLE
Tag latest not found in repository docker.io/library/hello-world

### ISSUE THREAD

**Issue by martin-wiseweb** [docker/docker#26295]
**Output of `docker version`:**

```
Client:
 Version:         1.10.3
 API version:     1.22
 Package version: docker-common-1.10.3-46.el7.centos.10.x86_64
 Go version:      go1.6.3
 Git commit:      d381c64-unsupported
 Built:           Thu Aug  4 13:21:17 2016
 OS/Arch:         linux/amd64

Server:
 Version:         1.10.3
 API version:     1.22
 Package version: docker-common-1.10.3-46.el7.centos.10.x86_64
 Go version:      go1.6.3
 Git commit:      d381c64-unsupported
 Built:           Thu Aug  4 13:21:17 2016
 OS/Arch:         linux/amd64
```

**Output of `docker info`:**

```
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 2
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-253:1-41943262-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 309.8 MB
 Data Space Total: 107.4 GB
 Data Space Available: 4.628 GB
 Metadata Space Used: 823.3 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Execution Driver: native-0.2
Logging Driver: journald
Plugins: 
 Volume: local
 Network: host bridge null
Kernel Version: 3.10.0-327.28.3.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 2
CPUs: 1
Total Memory: 1.907 GiB
--
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Registries: docker.io (secure)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. Run `docker run hello-world`

**Describe the results you received:**

```
Unable to find image 'hello-world:latest' locally
Trying to pull repository docker.io/library/hello-world ... 
Pulling repository docker.io/library/hello-world
Tag latest not found in repository docker.io/library/hello-world
docker: Tag latest not found in repository docker.io/library/hello-world.
See '/usr/bin/docker-current run --help'.
```

**Describe the results you expected:**

```
Unable to find image 'hello-world:latest' locally
    latest: Pulling from hello-world
    a8219747be10: Pull complete
    91c95931e552: Already exists
    hello-world:latest: The image you are pulling has been verified. Important: image verification is a tech preview feature and should not be relied on to provide security.
    Digest: sha256:aa03e5d0d5553b4c3473e89c8619cf79df368babd1.7.1cf5daeb82aab55838d
    Status: Downloaded newer image for hello-world:latest
    Hello from Docker.
    This message shows that your installation appears to be working correctly.

    To generate this message, Docker took the following steps:
     1. The Docker client contacted the Docker daemon.
     2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
            (Assuming it was not already locally available.)
     3. The Docker daemon created a new container from that image which runs the
            executable that produces the output you are currently reading.
     4. The Docker daemon streamed that output to the Docker client, which sent it
            to your terminal.

    To try something more ambitious, you can run an Ubuntu container with:
     $ docker run -it ubuntu bash

    For more examples and ideas, visit:
     http://docs.docker.com/userguide/
```

**Additional information you deem important (e.g. issue happens only occasionally):**

I'm unable to ping docker.com nor docker.io
```
ping docker.com
PING docker.com (54.84.88.245) 56(84) bytes of data.

^C
--- docker.com ping statistics ---
11 packets transmitted, 0 received, 100% packet loss, time 10008ms
```
```
# ping docker.io
PING docker.io (52.207.178.113) 56(84) bytes of data.

^C
--- docker.io ping statistics ---
10 packets transmitted, 0 received, 100% packet loss, time 9000ms
```

**Comment by martin-wiseweb** [comment]
Docker, upon installation, `docker run hello-world` always shows me `x509: certificate signed by unknown authority.`.

So I fix this by doing the following:

```
ln -s /etc/pki/tls/certs/ca-bundle.crt /etc/docker/certs.d/docker.io
ln -s /etc/pki/tls/certs/ca-bundle.trust.crt /etc/docker/certs.d/docker.io
```

Running `docker run hello-world` or `docker pull hello-world` produce the errors above at this point.

**Comment by mikeeus** [comment]
@martin-wiseweb Hi martin, I had the 'Tag latest not found' error as you did. I think it may be a firewall issue because when I added a proxy in the docker default file it was solved. For me the file was found at `etc/default/docker`.

I'm using Ubuntu 16.04 so the file path may be different for you, but give it a shot and gl.

**Comment by Selvaguru** [comment]
Hi,
I faced the same problem. Disabling the firewall did the trick !

**Comment by thaJeztah** [comment]
I suspect this is a configuration issue indeed; also, it looks like you're running a Red Hat fork of Docker, not the official build; the Red Hat fork has a lot of modifications in this area (and, for example, an option to actively block access to Docker Hub), so it may be worth installing the regular packages; https://docs.docker.com/engine/installation/linux/centos/

I'm going to close this issue, because this is not a bug, but feel free to continue the discussion

**Comment by MSaifAsif** [comment]
I fixed the problem by uograding to 1.9.1 from 1.8
```
[~]$ apt-get install docker-engine=1.9.1-0~trusty
```

**Comment by GNQ2018** [comment]
after I login dockerHub in kitematic , I can use docker client terminal .
